### PR TITLE
[Documentation][SyliusTaxationBundle] Docs improving taxation bundle

### PR DESCRIPTION
**Pull request by Mrkisha**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Related tickets |  |
| License | MIT |

I did not know how to update previous pull request (#6441). So, here comes this one.

### CODE REVIEW

File: docs/bundles/SyliusTaxationBundle/configuring_taxation.rst [hunk 1]
```diff
@@ -17,19 +17,19 @@ First step is to create a new tax category.
         $repository = $this->container->get('sylius.repository.tax_category');
         $manager = $this->container->get('sylius.manager.tax_category');
 
-        $clothing = $repository
+        $cloudServerTaxCategory = $repository
             ->createNew()
```
Comment (pamil) [inline docs/bundles/SyliusTaxationBundle/configuring_taxation.rst]: Ouch, we don't create entities in repositories for a long time, this should be updated to use factories.

Comment (Mrkisha) [inline docs/bundles/SyliusTaxationBundle/configuring_taxation.rst]: @pamil 
I've just used same API used in current docs.
What's the API for factory, so I can change it?

Comment (lchrusciel) [inline docs/bundles/SyliusTaxationBundle/configuring_taxation.rst]:
```php
$factory = $this->container->get('sylius.factory.tax_category');

$cloudServerTaxCategory = $factory->createNew();

$cloudServerTaxCategory->set();
$cloudServerTaxCategory->set();

$repository = $this->container->get('sylius.repository.tax_category');
$repository->add($cloudServerTaxCategory);
```

No entity manager needed. We have also removed fluent interfaces.

File: docs/bundles/SyliusTaxationBundle/configuring_taxation.rst [hunk 2]
```diff
@@ -45,52 +45,56 @@ Second thing to do is to classify the taxables, in our example we'll get two pro
...
     public function configureAction()
     {
-        $tshirt = // ...
-        $banana = // ... Some logic behind loading entities.
-
         $repository = $this->container->get('sylius.repository.tax_category');
 
-        $clothing = $repository->findOneBy(array('name' => 'Clothing'));
-        $food = $repository->findOneBy(array('name' => 'Food'));
+        $cloudServerTaxCategory = $repository->findOneBy(array('name' => 'Cloud Server'));
```
Comment (lchrusciel) [inline docs/bundles/SyliusTaxationBundle/configuring_taxation.rst]: `$repository->findOneBy(['name' => 'Cloud Server']);`

File: docs/bundles/SyliusTaxationBundle/configuring_taxation.rst [hunk 3]
```diff
@@ -45,52 +45,56 @@ Second thing to do is to classify the taxables, in our example we'll get two pro
...
+        $sharedHostingTaxCategory = $repository->findOneBy(array('name' => 'Shared Hosting'));
+
+        $digitalOcean = new Server();
+        $taxCategory->setName('Digital Ocean');
```
Comment (lchrusciel) [inline docs/bundles/SyliusTaxationBundle/configuring_taxation.rst]: Didn't you mean `$digitalOcean`, did you?

File: docs/bundles/SyliusTaxationBundle/configuring_taxation.rst [hunk 4]
```diff
@@ -45,52 +45,56 @@ Second thing to do is to classify the taxables, in our example we'll get two pro
...
+        $taxCategory->setTaxCategory($cloudServerTaxCategory);
+
+        $goDaddy = new Server();
+        $taxCategory->setName('GoDaddy');
+        $taxCategory->setTaxCategory($sharedHostingTaxCategory);
 
-        $tshirt->setTaxCategory($clothing);
-        $food->setTaxCategory($food);
+        $manager->persist($digitalOcean);
```
Comment (lchrusciel) [inline docs/bundles/SyliusTaxationBundle/configuring_taxation.rst]: If `Server` is a sylius resource it should also have a factory and then be added to database with `repository->add()`. Entity manager would not be needed

File: docs/bundles/SyliusTaxationBundle/configuring_taxation.rst [hunk 5]
```diff
@@ -45,52 +45,56 @@ Second thing to do is to classify the taxables, in our example we'll get two pro
...
-
-        $clothing = $taxCategoryRepository->findOneBy(array('name' => 'Clothing'));
-        $food = $taxCategoryRepository->findOneBy(array('name' => 'Food'));
+        //... continue
 
         $repository = $this->container->get('sylius.repository.tax_rate');
         $manager = $this->container->get('sylius.repository.tax_rate');
 
-        $clothingTax = $repository
+        $cloudServerTax = $repository
```
Comment (lchrusciel) [inline docs/bundles/SyliusTaxationBundle/configuring_taxation.rst]: Same as above. Add factory, remove entity manager

File: docs/bundles/SyliusTaxationBundle/custom_calculators.rst [hunk 6]
```diff
@@ -23,3 +23,28 @@ All calculators implement the **TaxCalculatorInterface**. First, you need to cre
...
+            ->setCalculator('app.server_tax')
+            ->setAmount(0,15);
+
+Service should be tagged with key value pair ``name: sylius.tax_calculator`` and ``calculator: app.server_tax``.
+
+.. code-block:: yaml
+
+    #
+    app.tax_calculator.server_tax_calculator:
+        class: AppBundle\Taxes\Calculator\TouristTaxCalculator
```
Comment (lchrusciel) [inline docs/bundles/SyliusTaxationBundle/custom_calculators.rst]: Can we make this example more consistent? A few lines above we have created a `FeeCalculator` and now we are registering a `FeeCalculator` under `app.server_tax` tag.

File: docs/bundles/SyliusTaxationBundle/custom_calculators.rst [hunk 7]
```diff
@@ -23,3 +23,28 @@ All calculators implement the **TaxCalculatorInterface**. First, you need to cre
             return $amount * ($rate->getAmount() + 0,15 * 0,30);
         }
     }
+
+
+.. note::
+
+    Now we need to register calculator as service.
+
+Remember we set calculator for tax rate as ``app.server_tax``.
```
Comment (lchrusciel) [inline docs/bundles/SyliusTaxationBundle/custom_calculators.rst]: This could go to the bottom of file as example usage

**Pull request closed without merging**